src/network/networkd-dhcp-common.c

```diff
@@ -63,6 +63,40 @@ int config_parse_dhcp(
         return 0;
 }
 
+int config_parse_dhcp_route_metric(
+                const char* unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        Network *network = data;
+        uint32_t metric;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = safe_atou32(rvalue, &metric);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r,
+                           "Failed to parse RouteMetric=%s, ignoring assignment: %m", rvalue);
+                return 0;
+        }
+
+        network->dhcp_route_metric = r;
+        network->dhcp6_route_metric = r;
```

Comment:
Hmm, that means that if specify
```
[DHCPv6]
RouteMetric=1
[DHCP]
RouteMetric=2
```
I'll get 2 for dhcp6_route_metric. Shouldn't this only be set if the other setting is not used?

Reply:
`[DHCP]` section is now deprecated. So, I think it is not necessary to consider the case that `[DHCP]` and `[DHCPv6]` sections are simultaneously used.

Reply:
Why we are setting the both ?

Reply:
> Why we are setting the both ?

For backwards compatibility. See the commit message.

> `[DHCP]` section is now deprecated. So, I think it is not necessary to consider the case that `[DHCP]` and `[DHCPv6]` sections are simultaneously used.

Actually, I think people might reasonably set both, to support newer and older systemd simultaneously.

Reply:
DHCPv6 route metric never supported and `[DHCP]` section is deprecated.

Reply:
"deprecated" means "supported, but not encouraged". So if something is deprecated, we still need handle things correctly.

Reply:
OK. I changed my mind. Now `[DHCPv4]` or `[DHCPv6]` sections take precedence. PTAL.